### PR TITLE
[REVIEW] docs, Dockerfile & read_csv improvements

### PR DESCRIPTION
**Pull request by randerzander**
Docs:
Fix example for `merge`, adding examples for more DataFrame and Series methods
Clean up read_csv and read_csv docs
Closes https://github.com/rapidsai/cudf/issues/275
Add CategoricalAccessor to doc pages

Dockerfile:
Added pytest conda dep
Switched nvstrings to "nvidia" conda channel
Switched to adding local source directories instead of cloning from master-
   Lets users build/test local changes to cudf Python bindings without rebuilding all of libgdf

cudf.io.read_csv -- start supporting 'sep' argument, update tests accordingly

### CODE REVIEW

**Comment by randerzander** [comment]
Not sure why libgdf's [GroupValidTest would fail](http://gpuci.gpuopenanalytics.com/job/cudf-prb/637/testReport/(root)/GroupValidTest_0/ReportValidMaskError/), as it's unrelated to this PR. Is this test broken in master?

**Comment by kkraus14** [comment]
@randerzander I saw this in an unrelated PR as well.

cc @jrhemstad any idea why this test would be failing?

**Comment by harrism** [comment]
@kkraus14 I'm unable to repro the test failure locally. And interestingly, it's only failing in this PR -- can you find any other tests from other PRs or on master where it is failing? 

@kaatish: you added this test recently, can you take a look?

**Comment by kkraus14** [comment]
rerun tests

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by kkraus14** [comment]
add to whitelist

File: cudf/dataframe.py [hunk 1]
```diff
@@ -279,7 +282,21 @@ def assign(self, **kwargs):
 
     def head(self, n=5):
         """
-        Returns the first n rows as a new DataFrame
+        Return the first n rows.
+
+        This function returns the first n rows for the object based
+        on position. It is useful for quickly testing if your
+        object has the right type of data in it.
```
Comment (harrism) [inline cudf/dataframe.py]:
```suggestion
        DataFrame has the right type of data in it.
```

File: cudf/dataframe.py [hunk 2]
```diff
@@ -279,7 +282,21 @@ def assign(self, **kwargs):
 
     def head(self, n=5):
         """
-        Returns the first n rows as a new DataFrame
+        Return the first n rows.
+
+        This function returns the first n rows for the object based
```
Comment (harrism) [inline cudf/dataframe.py]:
```suggestion
        This function returns the first n rows of the DataFrame based
```

File: cudf/dataframe.py [hunk 3]
```diff
@@ -279,7 +282,21 @@ def assign(self, **kwargs):
...
+        on position. It is useful for quickly testing if your
```
Comment (harrism) [inline cudf/dataframe.py]: What do you mean by "based on position"? Isn't it sufficient to say "... returns the first n rows of the DataFrame."?

File: cudf/dataframe.py [hunk 4]
```diff
@@ -279,7 +282,21 @@ def assign(self, **kwargs):
...
+        object has the right type of data in it.
+
+        Parameters
+        ----------
+          n : int, default 5
+            Number of rows to select.
```
Comment (harrism) [inline cudf/dataframe.py]:
```suggestion
            Number of rows to return.
```

File: cudf/dataframe.py [hunk 5]
```diff
@@ -279,7 +282,21 @@ def assign(self, **kwargs):
...
+
+        Returns
+        -------
+        obj_head : type of caller
```
Comment (harrism) [inline cudf/dataframe.py]: Why "type of caller"? This function returns a DataFrame with n rows.

File: cudf/dataframe.py [hunk 6]
```diff
@@ -279,7 +282,21 @@ def assign(self, **kwargs):
 
     def head(self, n=5):
         """
-        Returns the first n rows as a new DataFrame
+        Return the first n rows.
```
Comment (harrism) [inline cudf/dataframe.py]: Why remove "as a new DataFrame"? I think this is an important detail: head is implemented using slicing, which returns a DataFrame.

File: cudf/dataframe.py [hunk 7]
```diff
@@ -387,33 +404,48 @@ def loc(self):
...
+        """
+        The index (row labels) of the DataFrame.
         """
         return self._index
 
     def set_index(self, index):
-        """Return a new DataFrame with a new index
+        """
+        Set the DataFrame index (row labels) using one or more existing
+        columns. Yields a new object.
```
Comment (harrism) [inline cudf/dataframe.py]:
```suggestion
        columns. Yields a new DataFrame object.
```

File: cudf/io.py [hunk 8]
```diff
@@ -18,19 +18,25 @@ def _wrap_string(text):
...
+
     Parameters
     ----------
     filepath : str
         Path of file to be read.
-    delimiter : char, default ','
+    sep : str, default ','
         Delimiter to be used.
+    delimiter : str, default None
+        Alternative argument name for sep.
```
Comment (harrism) [inline cudf/io.py]: If these are alternatives, why do they have different defaults?

File: cudf/series.py [hunk 9]
```diff
@@ -698,7 +698,8 @@ def factorize(self, na_sentinel=-1):
     # UDF related
 
     def applymap(self, udf, out_dtype=None):
-        """Apply a elemenwise function to transform the values in the Column.
+        """
+        Apply an elemenwise function to transform the values in the Column.
```
Comment (harrism) [inline cudf/series.py]:
```suggestion
        Apply an element-wise function to transform the values in the Column.
```

**Comment by harrism** [comment]
@randerzander Are the changes in this PR related? Title sounds like 3 separate things -- suggest separate PRs in the future.

In any case, can you fix the conflicts and failures and review requests? Thanks!

**Pull request closed without merging**